model/task/db.go

```diff
@@ -788,25 +788,36 @@ func needsContainerAllocation() bson.M {
 //     have been met.
 //  2. The container is allocated but the agent has not picked up the task yet.
 func IsContainerTaskScheduledQuery() bson.M {
+	query := IsUndispatchedContainerTasksQuery()
+	query["$or"] = dependenciesAreMetQuery()
+	return query
+}
+
+// IsUndispatchedContainerTasksQuery returns a query returning all tasks that
+// are undispatched container tasks.
```

Comment:
nit: the phrasing is a little awkward, maybe something like "returns all undispatched container tasks"?